src/avram/attribute.cr

```diff
@@ -98,4 +98,64 @@ class Avram::Attribute(T)
       add_error("is invalid")
     end
   end
+
+  # These methods may accidentally get called on attributes
+  # inside of operations. Since these methods don't exist,
+  # chances are, you meant to call them on the value.
+  # ```
+  # username.to_s
+  # # VS
+  # username.value.to_s
+  # ```
+  def to_s
+    call_value_instead_error_message(".to_s")
+  end
+
+  # NOTE: to_s(io : IO) is used when passing an object
+  # in to string interpolation. Don't override that method.
+  def to_s(time_format : String)
+    call_value_instead_error_message(".to_s(time_format)")
+  end
```

Comment:
I think these are a great addition. If there's one mistake I make _all the time_ it's calling these methods accidentally. 

This specific overload makes me a bit nervous because of the way the name of the string parameter would confuse someone who happened upon this error message through some secondary path that had nothing to do with a time format. 

What if the stringified method suggestion in the error message is just `.to_s(...)` instead? That seems like it would lead people down an almost identical path without lending confusion where they landed here without doing anything with time formats.

Reply:
I like that 👍  Makes sense.